jax/_src/scipy/signal.py

```diff
@@ -147,11 +147,29 @@ def convolve(in1: Array, in2: Array, mode: str = 'full', method: str = 'auto',
 @implements(osp_signal.convolve2d)
 def convolve2d(in1: Array, in2: Array, mode: str = 'full', boundary: str = 'fill',
                fillvalue: float = 0, precision: PrecisionLike = None) -> Array:
-  if boundary != 'fill' or fillvalue != 0:
-    raise NotImplementedError("convolve2d() only supports boundary='fill', fillvalue=0")
+  if boundary not in ["fill", "wrap"]:
+    raise NotImplementedError("convolve2D() only supports boundary='fill' or 'wrap'")
+  elif boundary == 'fill' and fillvalue != 0:
+    raise NotImplementedError("convolve2d() only supports fillvalue=0 when boundary='fill'")
+
   if jnp.ndim(in1) != 2 or jnp.ndim(in2) != 2:
     raise ValueError("convolve2d() only supports 2-dimensional inputs.")
-  return _convolve_nd(in1, in2, mode, precision=precision)
+
+  if boundary == 'wrap':
```

Comment:
What if we put this logic in `_convolve_nd`, then `correlate2d` could make use of it as well?